src/network.rs

```diff
@@ -246,232 +224,286 @@ impl Nectar {
-                                shared_swap_id,
-                                remote_data
-                            );
-                            return;
-                        }
-                    };
-
-                let (maker_bitcoin_transient_sk, maker_ethereum_identity) =
-                    match self.local_identities.remove(&shared_swap_id) {
-                        Some(LocalIdentities { bitcoin, ethereum }) => (bitcoin, ethereum),
-                        _ => {
-                            tracing::warn!(
-                                "could not find identities for shared_swap_id: {}",
-                                shared_swap_id,
-                            );
-                            return;
-                        }
-                    };
-
-                let order = match self.order_ids.get(&shared_swap_id) {
-                    Some(order_id) => match self.orderbook.get_order(order_id) {
-                        Some(order) => order,
-                        None => {
-                            tracing::warn!("could not find order with id: {}", order_id);
-                            return;
-                        }
-                    },
-                    None => {
-                        tracing::warn!(
-                            "could not find order_id corresponding to shared_swap_id: {}",
-                            shared_swap_id
-                        );
-                        return;
-                    }
-                };
-
-                let taker = match self.takers.get(&order.id) {
-                    Some(taker_peer_id) => Taker::new(taker_peer_id.clone()),
-                    None => {
-                        tracing::warn!("unknown taker for order: {}", order.id,);
-                        return;
-                    }
-                };
-
-                let swap = match order.position {
-                    // Bob is buying (=redeeming) bitcoin (funding DAI on beta, redeeming bitcoin on alpha)
-                    comit::network::Position::Buy => {
-                        let redeem_identity = identity::Bitcoin::from_secret_key(
-                            &crate::SECP,
-                            &maker_bitcoin_transient_sk,
-                        );
-                        let hbit_params = hbit::Params::new(
-                            hbit::SharedParams {
-                                asset: order.bitcoin_amount,
-                                redeem_identity,
-                                refund_identity: taker_bitcoin_identity,
-                                expiry: order.bitcoin_absolute_expiry.into(),
-                                secret_hash,
-                                network: order.bitcoin_ledger.into(),
+            ::comit::network::setup_swap::BehaviourOutEvent::ExecutableSwap(exec_swap) => {
+                let swap_id = exec_swap.context;
+                // todo: get from order match ref point
```

Comment:
You can solve this TODO easily by introducing a context that contains the reference point. I did the same thing in cnd.